python/core/fundamental-data-structures/py-tuple-assignment.md

```diff
@@ -0,0 +1,114 @@
+---
+author: kapnobatai136
+
+tags:
+  - introduction
+
+type: normal
+
+category: must-know
+
+aspects:
+  - introduction
+
+links:
+  - '[Tuples and Sequences](https://docs.python.org/3.5/tutorial/datastructures.html#tuples-and-sequences){website}'
+
+---
+
+# Tuple Assignment
+
+---
+## Content
+
+The **tuple assignment** feature allows packing and unpacking such that a tuple left of the assignment will be assigned all values from the tuple right of the assignment:
+```python
+flash = ("Berry", "Allen" , 27)
+(fname, lname, age) = flash
+print(fname)
+# 'Berry'
+print(age)
+# 27
+```
+
+In case of tuples with multiple fields, use the `*` unpacking syntax:
+```python
+a, *b, c = (1, 2, 3, 4, 5)
+print(a)
+# 1
+print(b)
+# [2, 3, 4]
+print(c)
+# 5
+```
+
+> Note: only one starred expression (`*`) is allowed per assignment[1].
+
+In this example, we assign the first element of the tuple to `a`, the last element of the tuple to `c`, and everything in-between to `b`.
+
+Tuples are often used for a sequence of values of heterogeneous types[2] (e.g. fields of a database record or columns from a CSV file), while lists are often used for homogeneous items[3]. This is not a rule; it can be the other way around too.
+
+Some advantages of tuples include:
+- faster than lists (due to them being immutable)
```

Comment:
do we explain immutability prior to this insight?

Reply:
not sure, but I've removed this section